src/main/java/DukeException.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * DukeException class handles the exception specific to Duke chatbot.
+ */
+public class DukeException extends Exception {
+    public String tmp = "";
+    public int num = 0;
+    public String tmp2 = "";
+    public String tmp3 = "";
```

Comment:
If I am not wrong, I think that each string and integer will be initialised to "" and 0 respectively. So if you would like to, you can change it to just "public int num;". Furthermore, variable names can be more descriptive as tmp1 and tmp2 might become quite confusing.